File: src/main/java/com/github/cybellereaper/noblepets/core/talent/defaults/GuardianShellTalent.java
package com.github.cybellereaper.noblepets.core.talent.defaults;

import com.github.cybellereaper.noblepets.api.ActivePet;
import com.github.cybellereaper.noblepets.api.StatType;
import org.bukkit.Particle;

public final class GuardianShellTalent extends PeriodicTalent {
  public GuardianShellTalent() {
    super(240);
  }

  @Override
  public String getId() {
    return "guardian_shell";
  }

  @Override
  public String getDisplayName() {
    return "Guardian Shell";
  }

  @Override
  public String getDescription() {
    return "Periodically grants absorption hearts to protect you.";
  }

  @Override
  protected void trigger(ActivePet pet) {
    double absorption = 2.0 + pet.statValue(StatType.DEFENSE) * 0.05;
    pet.grantAbsorption(absorption);
    pet.getOwner()
        .getWorld()
        .spawnParticle(
            Particle.INSTANT_EFFECT, pet.getOwner().getLocation(), 20, 0.5, 0.5, 0.5, 0.0);
  }
}

File: src/main/kotlin/com/github/cybellereaper/wizpets/core/talent/defaults/GuardianShellTalent.kt
package com.github.cybellereaper.wizpets.core.talent.defaults

import com.github.cybellereaper.wizpets.api.ActivePet
import com.github.cybellereaper.wizpets.api.StatType
import com.github.cybellereaper.wizpets.api.talent.PetTalent
import org.bukkit.Particle

class GuardianShellTalent : PeriodicTalent(intervalTicks = 240), PetTalent {
    override val id: String = "guardian_shell"
    override val displayName: String = "Guardian Shell"
    override val description: String = "Periodically grants absorption hearts to protect you."

    override fun trigger(pet: ActivePet) {
        val absorption = 2.0 + pet.statValue(StatType.DEFENSE) * 0.05
        pet.grantAbsorption(absorption)
        pet.owner.world.spawnParticle(Particle.INSTANT_EFFECT, pet.owner.location, 20, 0.5, 0.5, 0.5, 0.0)
    }
}
